app/src/main/java/br/com/dcarv/arcmovies/domain/abstraction/ICountryMapper.kt

```diff
@@ -0,0 +1,10 @@
+package br.com.dcarv.arcmovies.domain.abstraction
+
+import br.com.dcarv.arcmovies.domain.model.Country
+
+/**
+ * @author Danilo Carvalho
+ */
+interface ICountryMapper<in T> {
+    fun toCountry(obj: T): Country
+}
```

Comment:
The reason for all the Mapper classes was to keep the model classes clean from Gson annotations or any other platform clutter.
For example, some of the properties returned by the API had underscore case ("release_date") instead of camel case ("releaseDate"). To keep the model properties consistent with Kotlin guidelines I'd have to either annotate these model properties with @SerializedName or create a custom Gson serializer.
Or if we decided on storing a cache locally in SQLite, for example, and we were using an ORM library, we wouldn't have to add annotations/code from this library into the model. We could create another class ("DbCountry", for example) which had all the ORM bindings, implement a mapper class to transform a Country instance into a DbCountry instance and use that to save to SQLite.

The mapper classes would also help avoid having to refactor parts of the app in case we switch to a different API (or different ORM). All business logic and presentation classes use the model class "Country", while the API calls use the "TMDBCountry". If we had to switch to another API that uses a different representation for countries, we would just have to create a new mapper for that new representation into "Country" and the rest of the app would remain intact.

Finally, the reason for the mapper INTERFACES is so we can reference them in business logic code (if we ever have to) without having to reference a specific mapper class and avoid refactoring.